ssl/ssl_lib.c

```diff
@@ -3823,6 +3823,169 @@ static int ssl_session_cmp(const SSL_SESSION *a, const SSL_SESSION *b)
     return memcmp(a->session_id, b->session_id, a->session_id_length);
 }
 
+SSL_CTX *SSL_CTX_dup(OSSL_LIB_CTX *libctx, SSL_CTX *source,
+                     const char *propq, const SSL_METHOD *meth)
+{
+    SSL_CTX *ret = NULL;
+    int ok = 0;
+
+    if (source == NULL)
+        goto exit;
+
+    ret = SSL_CTX_new_ex(libctx, propq, meth);
+    if (ret) {
+        if (meth != NULL)
+            ret->cnf->method = meth;
+        else
+            ret->cnf->method = source->cnf->method;
+        if (source->cnf->cipher_list) {
+            sk_SSL_CIPHER_free(ret->cnf->cipher_list);
+            ret->cnf->cipher_list = sk_SSL_CIPHER_dup(source->cnf->cipher_list);
```

Comment:
What is the plan for malloc failures here? Ignoring them is not really an option :)